### PR TITLE
removed printing configuration file content on stderr

### PR DESCRIPTION
**Pull request by Mararok**
Refers to #171

### CODE REVIEW

**Comment by p-alik** [comment]
Thank you, @Mararok.
@SpamapS,  does failed OSX build is an obstacle to merge this PR?

**Comment by esabol** [comment]
It seems to me that the ideal solution would be to “XXXXX” out the password argument(s) and still display the others.

**Comment by Mararok** [comment]
> It seems to me that the ideal solution would be to “XXXXX” out the password argument(s) and still display the others.

That solution will be better in commandline. In mysql they are replacing password value with xxx  in that line: http://bazaar.launchpad.net/~mysql/mysql-server/5.5/view/head:/client/mysql.cc#L1734.

Printing that options wasn't a real problem when you use "normal" log file, but in docker all output on stderr will be grabbed and saved.

**Comment by Mararok** [comment]
Now It's ok for all ?

File: gearmand/gearmand.cc [hunk 1]
```diff
@@ -264,12 +264,14 @@ int main(int argc, char *argv[])
         std::vector<std::string> args;
         std::copy(tok.begin(), tok.end(), back_inserter(args));
 
-        for (std::vector<std::string>::iterator iter= args.begin();
-             iter != args.end();
-             ++iter)
-        {
-          std::cerr << *iter << std::endl;
-        }
+      	#if defined(DEBUG) && DEBUG
```
Comment (p-alik) [inline gearmand/gearmand.cc]: all other `#if` directives are placed at the beginning of of line.
`git grep '#if' -- '*.cc'`
Could you replace `#if` and `#endif` at the line beginning, please.

**Comment by p-alik** [comment]
as expected:

- `./configure`
```bash
$ ./gearmand/gearmand --config-file gearmand.conf
./gearmand/gearmand: Could not open log file "/usr/local/var/log/gearmand.log", from "/opt/devel/Gearman/vagrant/debian/jessie64/gearmand", switching to stderr. (No such file or directory)
  DEBUG 2018-09-21 19:47:08.388223 [  main ] THREADS: 4 -> libgearman-server/gearmand.cc:263
```

- `./configure --enable-debug` (#176 an other issue with this regard) 

```bash
$ ./gearmand/gearmand --config-file gearmand.conf
--port=4731
--verbose=DEBUG
./gearmand/gearmand: Could not open log file "/usr/local/var/log/gearmand.log", from "/opt/devel/Gearman/vagrant/debian/jessie64/gearmand", switching to stderr. (No such file or directory)
  DEBUG 2018-09-21 19:19:14.928433 [  main ] THREADS: 4 -> libgearman-server/gearmand.cc:263
```

**Comment by p-alik** [comment]
@SpamapS, do you have any objection to merge this PR?

**Comment by SpamapS** [comment]
Thanks, that's a pretty good middle-of-the-road solution. Would still prefer the "XXXXX"'s, but I know that's a lot more work.

**Pull request merged**